src/libasr/pass/replace_symbolic.cpp

```diff
@@ -219,6 +223,83 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
             }
         }
     }
+
+    void visit_Print(const ASR::Print_t &x) {
+        std::vector<ASR::expr_t*> print_tmp;
+        SymbolTable* module_scope = current_scope->parent;
+        for (size_t i=0; i<x.n_values; i++) {
+            ASR::expr_t* value = x.m_values[i];
+
+            if (ASR::is_a<ASR::Var_t>(*value) && ASR::is_a<ASR::CPtr_t>(*ASRUtils::expr_type(value))) {
```

Comment:
I think we might encounter a problem with this. We somehow recognize the `value` as SymbolicExpression.

Reply:
Wow , I reported the same thing in the comment below :)

Reply:
I have two ideas:
1. Use Vec<std::string> and store all the variables converted from SymbolicExpression to CPtr. Here modify only those which are present in the vector.
2. Modify the Type CPtr to differentiate SymbolicExpressio (CPtr(SymbolicExpression)`) and other (CPtr())

@certilk Can you please provide suggestions on this?

Reply:
An example of how this is affecting the pass has been pointed in the comment below (https://github.com/lcompilers/lpython/pull/2239#discussion_r1281341804)

Reply:
I would do it like this: Let's have `std::set<ASR::expr_t*>` where we store all variables that are "ours", that is, they are symbolic. Then in the print statement, just check if the `value` is inside this set.